dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -673,6 +672,32 @@ public void testGetArchivedRelatedChild()
         }
     }
 
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * Given Scenario: Try to check-in a version of an archived page with {@link Contentlet#DONT_VALIDATE_ME} flag on
+     * ExpectedResult: The method should not throw a {@link NullPointerException}, but a {@link DotDataException} because
+     * we are trying to check-in a version of an archived content
+     */
+    @Test
+    public void testCheckInArchivedPageShouldThrowDotDataException() {
+        final Contentlet contentlet = TestDataUtils
+                .getPageContent(true, APILocator.getLanguageAPI().getDefaultLanguage().getId());
+        ContentletDataGen.archive(contentlet);
+
+        final Contentlet newVersion = ContentletDataGen.checkout(contentlet);
+        newVersion.setBoolProperty(Contentlet.DONT_VALIDATE_ME, true);
+        try {
+            ContentletDataGen.checkin(newVersion);
+            fail();
+        } catch (Exception e){
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5907835)